### PR TITLE
separation of modules (rpc class)

### LINKED ISSUE #42: RPC Endpoint/Contract Updates
- [ ] Fix issues with Contract class
- [ ] Put Contract in it's own module
- [ ] Rename test_utils.py to test_rpc_endpoint.py

### PR DESCRIPTION
**Pull request by tallywiesenberg**
Closes #42 

`tox -e style` will fail b/c docstrings are at 79.7%. However, the RPCEndpoint class and the Contract class are covered.

### CODE REVIEW

**Comment by oraclown** [comment]
I think we should use the abi format in the btc-usd plugins example, but just move it to utils, since the tests and plugin need to use the same file. Working on a commit for this now..

File: src/telliot/utils/contract.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    #: ABI specifications of contract
+    abi: List[Dict[str, Any]]
+
+    def connect(self) -> web3.contract.Contract:
+        """Connect to EVM contract through an RPC Endpoint"""
+        if self.node.web3 is None:
+            raise TypeError("node is not connected")
+        else:
+            self.web3_contract = self.node.web3.eth.contract(
```
Comment (clamdad) [inline src/telliot/utils/contract.py:37]: Should fix.  Flagged as an issue in my editor: Adding a new instance attribute outside of the dataclass __init__ function.  Surprised mypy doesn't flag this.

File: src/telliot/utils/contract.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                address=self.address, abi=self.abi
+            )
+            return self.web3_contract
```
Comment (clamdad) [inline src/telliot/utils/contract.py]: Usages of this function don't use the returned connection.   Choose to return None, or True/False for connection result.

**Comment by clamdad** [comment]
Question: What happens to Contract.connect() when the RPCEndpoint has a valid web3 object, but it is not connected?

**Review by clamdad — APPROVED** [review]

File: src/telliot/utils/contract.py (hunk 2 above)
Comment (clamdad) [reply src/telliot/utils/contract.py]: Fixed

File: src/telliot/utils/contract.py (hunk 1 above)
Comment (clamdad) [reply src/telliot/utils/contract.py:37]: Fixed

Comment (tallywiesenberg) [reply src/telliot/utils/contract.py:37]: Aha, I didn't realize all class attributes should be detailed in the dataclass init

**Comment by tallywiesenberg** [comment]
> Question: What happens to Contract.connect() when the RPCEndpoint has a valid web3 object, but it is not connected?

RPCEndpoint.connect() is for instantiating a valid web3 object from the url, so if a valid web3 object is passed through the constructor, than it should not affect the connected method. I think RPCEndpoint.connect() should return True if the web3 object is valid.

**Comment by tallywiesenberg** [comment]
I appended a new Base class that standardizes our configuration of the pydantic BaseModel. closes #51

**Comment by oraclown** [comment]
Removed base class, because Tally will open a separate PR for that stuff. Reverted commits [7c42512](https://github.com/tellor-io/pytelliot/pull/43/commits/7c425122589432cee465d63ffdcf82fde46d3aca), [13408b8](https://github.com/tellor-io/pytelliot/pull/43/commits/13408b8c41635e029b3ed63c4cc3e49f385e4103), and[ bd998c0](https://github.com/tellor-io/pytelliot/pull/43/commits/bd998c0b50850489f60a77c20a76df4d39680818).

**Pull request merged**